src/components/Chat/MessageList/index.js

```diff
@@ -81,52 +87,69 @@ const MessageList = props => {
   useEffect(() => {
     // TODO: implement some kind of button to scroll down when there are new messages
     ref.current.scrollTop = ref.current.scrollHeight
+    if (document.getElementById('newMessages'))
+      document.getElementById('newMessages').scrollIntoView()
   })
 
+  let counter = 0
+
+  useEffect(() => {
+    const filtered = posts.filter(
+      p =>
+        p.type !== 'system_purpose_change' &&
+        (p.type === '' ||
+          ((p.type === 'system_join_channel' ||
+            p.type === 'system_leave_channel' ||
+            p.type === 'system_join_team' ||
+            p.type === 'system_leave_team') &&
+            !isAdmin(p.user_id)))
+    )
+    setFilteredPosts(filtered)
+  }, [posts])
+
   return (
-    <section className="chat-message-list-container chat--message-list" ref={ref}>
+    <section
+      className="chat-message-list-container chat--message-list"
+      ref={ref}
+    >
       <div className="chat--message-list--container">
         {posts.length > 0 &&
-          posts
-            .filter(
-              p =>
-                p.type !== 'system_purpose_change' &&
-                (p.type === '' ||
-                  ((p.type === 'system_join_channel' ||
-                    p.type === 'system_leave_channel' ||
-                    p.type === 'system_join_team' ||
-                    p.type === 'system_leave_team') &&
-                    !isAdmin(p.user_id)))
-            )
-            .map(post => {
-              const timestampValues = setTimeStampValues(post)
-              return (
-                post &&
-                post.user_id && (
-                  <Message
-                    key={post.id}
-                    id={post.id}
-                    files={post.file_ids}
-                    type={post.type}
-                    url={post.url}
-                    sender={getNickNamebyId(post.user_id)}
-                    text={post.message}
-                    senderId={post.user_id}
-                    currentUserId={currentUserId}
-                    directChannel={directChannel}
-                    timeSent={timestampValues.sendTime}
-                    dateSent={timestampValues.sendDate}
-                    showDate={timestampValues.show}
-                    channelId={channelId}
-                    senderMmUsername={getUsernameById(post.user_id, profiles)}
-                    iconMemberStatus={getIconMemberStatus(post.user_id)}
-                    isAdmin={isAdmin(post.user_id)}
-                    pinPost={pinPost}
-                    filesData={filesData}
-                  />
-                )
+          filteredPosts.map(post => {
+            const timestampValues = setTimeStampValues(post)
+            counter += 1
```

Comment:
I think there is couple of other ways to do this. At least there is timestamp for last message in channel (called `last_post_at`  which could be used to decide if the message is last one. 

I also noticed, that there are couple of interesting selectors in mattermost-redux, for example one called `getUnreadPostsChunk`. I think that returns those posts that are unread for specific channel.

Reply:
I think there might be a problem with using last_post_at here since it also applies to the posts that are not rendered in message-component (like channel purpose updates).

I tried using the getUnreadPostsChunk but I have to say I'm not sure if I'll manage it in this time.. I don't really know how to use a selector here, but I can try. If I could use it, and it would return all the unread posts, I guess it should work by filtering out the same system messages that are filtered out for message list in general.

Reply:
Ok, I will also take a look if the selector is easy to implement or not. But if that does not work the way it should, then let's just go with the counter :)